keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,866 @@
+  This through this both in small and large cases, again with respect to the
+  [supported limits].
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+[existing SLIs/SLOs]: https://git.k8s.io/community/sig-scalability/slos/slos.md#kubernetes-slisslos
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+What do preexisting readers of the field do?
+They are not aware that they should ignore the seconds and use millis.
+To address this, we should set whole second probe value to the minimum allowed if milliseconds is set.
+
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Combine old and new fields
+
+Allow for 0 second Probe, with setting millisecond.
+
+Cons:
+Loosens validation.
+
+### v2 api for probe.
+
+I think this means a v2 API for Container therefore Pod.
+This seems too invasive.
+
+All Seconds fields become resource.Quantity instead of int32. This supports subdivision in a single field.
+
+### OffsetMilliseconds
```

Comment:
We could frame this or something like it as an “early probe failure offset”: if set, the probe is allowed fail up to _n_ ms before it normally would (probes can always succeed early).

It would be nice to clamp this value to a maximum of 999 at admission time; 1000 ms or more being something you should achieve through the seconds field.

Reply:
pinged you on slack.. let's try to setup a call if you have some time to spare .. cheers

Reply:
I'm happy to consider this as addressed.